addons/l10n_de/models/ir_attachment.py

```diff
@@ -24,11 +24,23 @@ def _except_audit_trail(self):
         for attachment in audit_trail_attachments:
             move = id2move.get(attachment.res_id)
             if move and move.posted_before and move.country_code == 'DE':
-                raise UserError(_("You cannot remove parts of the audit trail."))
+                ue = UserError(_("You cannot remove parts of the audit trail."))
+                ue._audit_trail = True
+                raise ue
 
     def write(self, vals):
         if vals.keys() & {'res_id', 'res_model', 'raw', 'datas', 'store_fname', 'db_datas'}:
-            self._except_audit_trail()
+            try:
+                self._except_audit_trail()
+            except UserError as e:
+                if (
+                    not hasattr(e, '_audit_trail')
+                    or vals.get('res_model') == 'documents.document'
+                    or vals.keys() & {'raw', 'datas', 'store_fname', 'db_datas'}
+                ):
+                    raise  # do not raise if trying to version the attachment through a document
+                vals.pop('res_model', None)
+                vals.pop('res_id', None)
```

Comment:
> (a `UserErrorAuditTrail(UserError)` class would also help readability)

The web client doesn't support superclasses, so it would get displayed as a server error (traceback) instead of a user error.

Reply:
> We would recommend adding a test to make sure that this touchy flow is fixed and does not go beyond what is needed.

Sure, but how/in which module?
There is not module that has both `documents` and `l10n_de` as dependencies, and I don't think having a bridge just for that test is worth

Reply:
Indeed, and as `ValidationError` is a `UserError`, I guess it makes sense that it's not just a matter of using `isinstance`.

It works if re-raising with `raise UserError(e.args[0]) from e` though.

If this code is specific to `l10_de` with `documents` because only then we expect cases where we shouldn't raise, shouldn't this code be in a bridge (with the test)? There already are several bridges between documents and localizations. Your comment said " ... through a document". But anyway that too would be in master. 

> add the test with a skip when it is not installed

This would work for stable, in this module then.

Reply:
@flch-odoo I have added a test, but I don't think that your suggestions are easily feasible.
The first one, already discussed because the web client wouldn't understand. Your suffestion to re-raise doesn't work because there is no way of doing that within the `api.ondelete` too
The second one would make it harder to prefetch all the `account.move` related to the attachments like done in `_except_audit_trail`. Otherwise we would need to return a mapping between attachments and whether they should raise, which is even dirtier IMO.